src/zimlet/com_zimbra_tos/com_zimbra_terms_services.js

```diff
@@ -0,0 +1,109 @@
+ * for the Original Developer. In addition, Exhibit A has been modified to be consistent with Exhibit B.
+ *
+ * Software distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTY OF ANY KIND, either express or implied.
+ * See the License for the specific language governing rights and limitations under the License.
+ * The Original Code is Zimbra Open Source Web Client.
+ * The Initial Developer of the Original Code is Zimbra, Inc.  All rights to the Original Code were
+ * transferred by Zimbra, Inc. to Synacor, Inc. on September 14, 2015.
+ *
+ * All portions of the code are Copyright (C) 2011, 2013, 2014, 2016 Synacor, Inc. All Rights Reserved.
+ * ***** END LICENSE BLOCK *****
+ */
+
+/**
+ * Defines the Zimlet handler class.
+ *
+ */
+
+Com_Zimbra_Tos = function () {
+	if (this instanceof Com_Zimbra_Tos) {
+		Com_Zimbra_Tos = this;
+	}
+};
+
+/**
+ * Makes the Zimlet class a subclass of ZmZimletBase.
+ *
+ */
+
+Com_Zimbra_Tos.prototype = new ZmZimletBase();
+
+/**
+* This method gets called by the Zimlet framework when the zimlet loads.
+*
+*/
+Com_Zimbra_Tos.prototype.init = function() {
+	this._showPreferenceDlg();
+
+};
+
+/**
+ * Display Terms Of Services Dialog.
+ *
+ */
+Com_Zimbra_Tos.prototype._showPreferenceDlg =
+ function() {
+	//if zimlet dialog already exists...
+	if (this._preferenceDialog) {
+		this._preferenceDialog.popup();
+		return;
+	}
+	this._preferenceView = new DwtComposite(this.getShell());
+	this._preferenceView.getHtmlElement().style.overflow = "auto";
+	this._preferenceView.getHtmlElement().innerHTML = this._createPrefView();
+	this._preferenceDialog = this._createDialog({title: this.getMessage("TosZimlet_label"), view: this._preferenceView, standardButtons: [DwtDialog.OK_BUTTON], id: "TosZimletPref"});
+	this._preferenceDialog.setButtonListener(DwtDialog.OK_BUTTON, new AjxListener(this, this._okPreferenceBtnListener));
+
+	
+	this.acceptedtos = Boolean(this.getUserProperty("zimbraPrefAcceptedClientTOS"));
+
+	document.getElementById("tos_accept_div1").checked = this.acceptedtos;
+	if (!this.acceptedtos) {
+		this._preferenceDialog.popup();
+	}
+ }
+ 
+/**
+ * Creates preferences view
+ */
+Com_Zimbra_Tos.prototype._createPrefView =
+function() {
+	return [
+		"<div class='toscontent'>",
+		this.getMessage("TosZimlet_content"),
+		"</div>",
+		"<div class='tosDialog'>",
+		"<input id='tos_accept_div1' type='checkbox'/>",
+		this.getMessage("TosZimlet_description"),
+		"</div>"
+	].join("");
+};
+
+//Zimlet framework calls this when the overview panel icon is single clicked
+Com_Zimbra_Tos.prototype.singleClicked =
+function() {
+	this._showPreferenceDlg();
+};
+
+/**
+ * Listens for the OK preferences button.
+ * 
+ */
+Com_Zimbra_Tos.prototype._okPreferenceBtnListener =
+function() {
+	this._preferenceDialog.popdown();
+	var domVal = document.getElementById("tos_accept_div1").checked;
+	this.setUserProperty("zimbraPrefAcceptedClientTOS", domVal, true);
```

Comment:
`this.setUserProperty("zimbraPrefAcceptedClientTOS", domVal, true);` doesn't update the account setting zimbraPrefAcceptedClientTOS defined in https://github.com/Zimbra/zm-mailbox/pull/858.
It updates zimbraZimletUserProperties like 

    $ zmprov ga test1 zimbraZimletUserProperties
    # name test1@dev.zimbra.com
    zimbraZimletUserProperties: com_zimbra_tos:zimbraPrefAcceptedClientTOS:true

Is it expected behavior?
If so, we may be able to remove the definition of zimbraPrefAcceptedClientTOS from https://github.com/Zimbra/zm-mailbox/pull/858.
Please make the design clear.

Reply:
Yes, I am using ZimletUserProperty here not the ldap attribute from https://github.com/Zimbra/zm-mailbox/pull/858. This attribute is added to 8.8 backend so that it can work with https://github.com/ZimbraOS/zm-x-web/pull/970 (zimbra x frontend)

I will rename zimletuserproperty to avoid confusion